src/Fmt_ast.ml

```diff
@@ -2766,20 +2766,27 @@ and fmt_module_type c ({ast= mty} as xmty) =
       }
 
 and fmt_signature c ctx itms =
-  let grps =
-    List.group itms ~break:(fun itmI itmJ ->
-        let is_simple itm =
-          match itm.psig_desc with
-          | Psig_open _ -> true
-          | Psig_module {pmd_type= {pmty_desc= Pmty_alias _}} -> true
-          | _ -> false
-        in
-        (not (is_simple itmI)) || (not (is_simple itmJ)) )
-  in
-  let fmt_grp itms =
-    list itms "@\n" (sub_sig ~ctx >> fmt_signature_item c)
-  in
-  hvbox 0 (list grps "\n@;<1000 0>" fmt_grp)
+  hvbox 0
+    (list_pn itms (fun ?prev itm ?next:_ ->
+         let nl =
+           match prev with
+           | None -> fmt ""
+           | Some {psig_loc} ->
+               let dist =
+                 itm.psig_loc.loc_start.pos_lnum - psig_loc.loc_end.pos_lnum
+               in
+               if
+                 dist <= 1
+                 && List.is_empty
+                      (Source.tokens_at c.source
+                         { psig_loc with
+                           loc_start= psig_loc.loc_end
+                         ; loc_end= itm.psig_loc.loc_start } ~filter:
+                         (fun _ -> true ))
+               then fmt "@\n"
+               else fmt "\n@\n"
```

Comment:
Could we have a comment or descriptively-named function indicating what this test is looking for?